homeassistant/components/weather/__init__.py

```diff
@@ -0,0 +1,74 @@
+"""
+homeassistant.components.weather
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Weather component that handles meteorological data for your location.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/weather.html
+"""
+import logging
+
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import STATE_UNKNOWN
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = []
+
+DOMAIN = "weather"
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+# ATTR_WEATHER_CONDITION = 'weather_condition'
+# ATTR_WEATHER_TEMPERATURE = 'weather_temperature'
+# ATTR_WEATHER_APPARENT_TEMPERATURE = 'weather_apparent_temperature'
+# ATTR_WEATHER_HUMIDITY = 'weather_humidity'
+# ATTR_WEATHER_PRESSURE = 'weather_pressure'
+# ATTR_WEATHER_WIND_SPEED = 'weather_wind_speed'
+# ATTR_WEATHER_WIND_BEARING = 'weather_wind_bearing'
+# ATTR_WEATHER_CLOUD_COVERAGE = 'weather_cloud_coverage'
+# ATTR_WEATHER_RAIN = 'weather_rain'
+# ATTR_WEATHER_SNOW = 'weather_snow'
+# ATTR_WEATHER_SUMMARY = 'weather_summary'
+# ATTR_WEATHER_ICON = 'weather_icon'
+# ATTR_WEATHER_DEW_POINT = 'weather_dev_point'
+# ATTR_WEATHER_VISIBILITY = 'weather_visibility'
+# ATTR_WEATHER_NEAREST_STORM_DISTANCE = 'weather_nearest_storm_distance'
+# ATTR_WEATHER_NEAREST_STORM_BEARING = 'weather_nearest_storm_bearing'
+# ATTR_WEATHER_PRECIP_TYPE = 'weather_precip_type'
+# ATTR_WEATHER_PRECIP_INTENSITY = 'weather_precip_intensity'
+# ATTR_WEATHER_PRECIP_PROBABILITY = 'weather_precip_probability'
+# ATTR_WEATHER_OZONE = 'ozone'
+
+
+def setup(hass, config):
+    """ Setup the weather service. """
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    component.setup(config)
+
+    return True
+
+
+# pylint: disable=no-self-use
+class WeatherCondition(Entity):
+    """ Represents a weather condition. """
+
+    @property
+    def state(self):
```

Comment:
Instead of having the state be a property returning `STATE_UNKNOWN`, we should have it return `self.weather_summary`.

`weahter_summary`, should just be as other potential weather properties be defined on this class.

```python
@property
def state(self):
    return self.weather_summary or STATE_UNKNOWN

@property
def weather_summary(self):
    return None

@property
def weather_visibility(self):
    return None
```